forte/processors/data_augment/algorithms/sampler.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from abc import abstractmethod, ABC
+from typing import Dict, List, Optional
+import nltk
+from nltk.corpus import words, stopwords
+
+
+__all__ = [
+    "Sampler",
+    "UniformSampler",
+    "UnigramSampler",
+]
+
+
+class Sampler(ABC):
+    r"""
+    A base sampler, an abstract class.
+    """
+    def __init__(self):
+        random.seed()
+
+    @abstractmethod
+    def sample(self) -> str:
+        raise NotImplementedError
+
```

Comment:
The sampler design seems restricted.

For a uniform sampler, it should probably simply take a word list, instead of hardcoding with 'nltk'. And after all, 'nltk' is not a distribution.

Similarly, the `UniformSampler` can also simply take a `Dict`.  We don't have to assume a file containing these counts.